edx_when/api.py

```diff
@@ -281,7 +299,11 @@ def set_date_for_block(course_id, block_id, field, date_or_timedelta, user=None,
                     date_or_timedelta
                 )
                 existing_date.policy = models.DatePolicy.objects.get_or_create(**date_kwargs)[0]
-        existing_date.save()
+                needs_save = True
```

Comment:
As above, don't write `existing_date.policy` unless something has changed.

Reply:
This is already done in the current code above:
```
            if (
                existing_date.policy.abs_date != date_or_timedelta and
                existing_date.policy.rel_date != date_or_timedelta
            ):
```

Reply:
Ah, right.